internal/provider/kubernetes/controller.go

```diff
@@ -1543,132 +1548,143 @@ func (r *gatewayAPIReconciler) enqueueClass(_ context.Context, _ client.Object)
 	}}}
 }
 
-func (r *gatewayAPIReconciler) hasManagedClass(ep *egv1a1.EnvoyProxy) bool {
-	// The EnvoyProxy must be in the same namespace as EG.
-	if ep.Namespace != r.namespace {
-		r.log.Info("envoyproxy namespace does not match Envoy Gateway's namespace",
-			"namespace", ep.Namespace, "name", ep.Name)
-		return false
+// processGatewayParamsRef processes the infrastructure.parametersRef of the provided Gateway.
+func (r *gatewayAPIReconciler) processGatewayParamsRef(ctx context.Context, gtw *gwapiv1.Gateway, resourceMap *resourceMappings, resourceTree *gatewayapi.Resources) error {
+	if gtw == nil || gtw.Spec.Infrastructure == nil || gtw.Spec.Infrastructure.ParametersRef == nil {
+		return nil
 	}
 
-	gcList := new(gwapiv1.GatewayClassList)
-	err := r.client.List(context.TODO(), gcList)
-	if err != nil {
-		r.log.Error(err, "failed to list gatewayclasses")
-		return false
+	if resourceTree.EnvoyProxyForGatewayClass != nil && resourceTree.EnvoyProxyForGatewayClass.Spec.MergeGateways != nil && *resourceTree.EnvoyProxyForGatewayClass.Spec.MergeGateways {
+		return fmt.Errorf("infrastructure.parametersref must be nil when MergeGateways feature is enabled by gatewayclass")
```

Comment:
this can't be an error message, it will disrupt processing for the rest of the resources tied to other users

Reply:
nvm you are logging on caller side